Packet++/src/PacketUtils.cpp

```diff
@@ -49,14 +55,16 @@ uint16_t computeChecksum(ScalarBuffer<uint16_t> vec[], size_t vecSize)
 	{
 		sum = (sum & 0xffff) + (sum >> 16);
 	}
-
 	PCPP_LOG_DEBUG("Sum before invert = " << sum << ", 0x" << std::uppercase << std::hex << sum);
 
-	sum = ~sum;
+	// To obtain the checksum we take the ones' complement of this result
+	uint16_t result = sum;
+	result = ~result;
 
-	PCPP_LOG_DEBUG("Calculated checksum = " << sum << ", 0x" << std::uppercase << std::hex << sum);
+	PCPP_LOG_DEBUG("Calculated checksum = " << sum << ", 0x" << std::uppercase << std::hex << result);
 
-	return ((uint16_t) sum);
+	// We return the result in Network byte order
+	return htons(result);

```

Comment:
isn't there more change needed in the calling code if you change the result of this function?
I mean different output now on LE (the arch where this lib is most used on)

Reply:
the return was already returned in Network byte order:
I have removed the following line
`localSum = be16toh(localSum);`